app/models/concerns/creatable.rb

```diff
@@ -0,0 +1,216 @@
+      self.creatorships.build(pseud: pseud, enable_notifications: true)
+    end
+  end
+
+  # Update which of User.current_user's pseuds should be listed on the byline
+  # after saving.
+  def current_user_pseud_ids=(ids)
+    return unless User.current_user.is_a?(User)
+
+    @current_user_pseuds = Pseud.where(id: ids).to_a
+  end
+
+  # This behaves very similarly to new_bylines=, but because it's designed to
+  # be used for bulk editing works, it doesn't handle ambiguous pseuds well. So
+  # we need to manually refine our guess as much as possible.
+  def pseuds_to_add=(pseud_names)
+    names = pseud_names.split(",").reject(&:blank?).map(&:strip)
+
+    names.each do |name|
+      possible_pseuds = Pseud.parse_byline(name)
+
+      if possible_pseuds.size > 1
+        possible_pseuds = Pseud.parse_byline(name, assume_matching_login: true)
+      end
+
+      pseud = possible_pseuds.first
+
+      if pseud
+        creatorship = creatorships.find_or_initialize_by(pseud: pseud)
+        creatorship.enable_notifications = true
+      end
+    end
+  end
+
+  ########################################
+  # USEFUL FUNCTIONS
+  ########################################
+
+  # Update the pseuds on this item so that User.current_user's pseuds are
+  # replaced by the passed-in array of pseuds new_pseuds. If it's a Series, we
+  # also update the user's byline on any owned works in the series. If it's a
+  # Work, we also update the user's byline on any owned chapters in the series.
+  def set_current_user_pseuds(new_pseuds)
+    return unless User.current_user.is_a?(User)
+
+    user_id = User.current_user.id
+
+    children = if is_a?(Work)
+                 chapters.to_a
+               elsif is_a?(Series)
+                 works.to_a
+               else
+                 []
+               end
+
+    transaction do
+      children.each do |child|
+        next unless child.users.include?(User.current_user)
+        child.set_current_user_pseuds(new_pseuds)
+      end
+
+      # Create before destroying, so that we don't run into issues with
+      # deleting the very last creator.
+      new_pseuds.each do |pseud|
+        creatorships.approve_or_create_by(pseud: pseud)
+      end
+
+      creatorships.each do |creatorship|
+        creatorship.destroy unless new_pseuds.include?(creatorship.pseud) ||
+                                   creatorship.pseud&.user_id != user_id
+      end
+    end
+  end
+
+  # Figure out which creatorships will exist after saving.
+  #
+  # Excludes creatorships with a missing pseud, because those orphaned
+  # creatorships can break various bits of code if they're considered valid.
+  def creatorships_after_saving
+    creatorships.select(&:valid?).reject(&:marked_for_destruction?).
+      reject { |creatorship| creatorship.pseud.nil? }
+  end
+
+  # Calculate what the pseuds on this work will be after saving, taking into
+  # account validity, approval, and @current_user_pseuds.
+  def pseuds_after_saving
```

Comment:
Just going to throw this out for discussion: `creatorships_after_saving` is looking at all creatorships, regardless of approval status, but this method is only looking at pseuds from approved creatorships. Should it maybe be called `approved_pseuds_after_saving` for clarity? (Also kind of future-proofing, in case we ever need/want a method that gets all pseuds after saving regardless of approval status.)

Reply:
The `pseuds` association only includes pseuds from approved creatorships. So I picked the name `pseuds_after_saving` because the function calculates which pseuds will be listed in the `pseuds` association after the creation is saved. (Which is similar to the behavior of `creatorships_after_saving`.)

Reply:
Makes sense! 👍